low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -0,0 +1,411 @@
+            elif state == "deactivating" or state == "inactive":
+                self.not_active_services[service] = int(time.time())
+            elif state == "failed":
+                self.failed_services.append(service)
+                alert_index = 1
+            else:
+                alert_index = 4
+        elif prev_state == "deactivating":
+            if state == "inactive" or state == "activating":
+                if service not in self.not_active_services:
+                    self.not_active_services[service] = int(time.time())
+            elif state == "failed":
+                self.failed_services.append(service)
+                alert_index = 1
+            else:
+                alert_index = 4
+        elif prev_state == "inactive" or prev_state == "failed":
+            if state == "activating":
+                if service not in self.not_active_services:
+                    self.not_active_services[service] = int(time.time())
+            elif state == "active":
+                if service in self.failed_services:
+                    self.failed_services.remove(service)
+                    alert_index = 3
+                elif service in self.not_active_services:
+                    self.not_active_services.pop(service)
+            else:
+                alert_index = 4
+        elif prev_state == "activating":
+            if service in self.not_active_services:
+                self.not_active_services.pop(service)
+            if state == "inactive" or state == "deactivating":
+                self.failed_services.append(service)
+                alert_index = 1
+            elif state == "active":
+                if service in self.failed_services:
+                    self.failed_services.remove(service)
+                    alert_index = 3
+                else :
+                    # its a restart.
+                    pass
+            elif state == "failed":
+                self.failed_services.append(service)
+                alert_index = 1
+            else:
+                alert_index = 4
+
+        logger.debug(f"alert_index : {alert_index}")
+        if not self.not_active_services:
+            logger.debug("Not_active_services is EMPTY")
+        for s,t in self.not_active_services.items():
+            logger.debug(f"{s} : {t}, {int(time.time())}")
+        logger.debug(f"{self.failed_services}")
+
+        if alert_index != -1:
+            self.raise_alert(service, prev_state, state,
+                prev_substate, substate, prev_pid, pid,
+                alert_index)
+
+    def raise_alert(self, service, prev_state, state, prev_substate,
+                    substate, prev_pid, pid, alert_index):
+        """ Send the alert to ServiceMsgHandler."""
+        # Each alert info contains 4 fields
+        # 1.Description | 2.Alert Type | 3.Impact | 4.Recommendation
+        alert_info = [
+            [f"{service} is not found/running initially",   #index 0
```

Comment:
Why we need to differentiate where service not running **initially** or **later**. Message should consistent for a state transition.

Reply:
I just wanted to suggest that initially before even registering the service to monitoring the either the service unit doesn't exists or it is not in active state.  And was never previously in the active state that's why `initially`. But if that is not required we can remove this message and use message at index 1

Reply:
removed this alert message at index 0 .